### PR TITLE
Feature/prompt tuning

### PR DESCRIPTION
**Pull request by yh351016**
main dev merge feature/prompt tuning dev

### CODE REVIEW

File: models/ofa/unify_transformer.py [hunk 1]
```diff
@@ -1070,7 +1238,7 @@ def build_output_projection(self, args, dictionary, embed_tokens):
             self.layers.insert(((i+1) * args.decoder_layers) // (num_base_layers + 1), BaseLayer(args))
 
     def build_decoder_layer(self, args, no_encoder_attn=False, drop_path_rate=0.0):
-        layer = TransformerDecoderLayer(args, no_encoder_attn, drop_path_rate=drop_path_rate)
+        layer = TransformerDecoderLayer(args, no_encoder_attn, drop_path_rate=drop_path_rate,use_adapter=getattr(args, "adapter", False),adapter_dim=getattr(args, "adapter_dim", 200))
```
Comment (JustinLin610) [inline models/ofa/unify_transformer.py:1241]: pay attention to the format

File: models/ofa/unify_transformer_layer.py [hunk 2]
```diff
@@ -64,10 +122,13 @@ class TransformerEncoderLayer(nn.Module):
         args (argparse.Namespace): parsed command-line arguments
     """
 
-    def __init__(self, args, drop_path_rate=0.0):
+    def __init__(self, args, drop_path_rate=0.0,use_adapter=False,adapter_dim=200):
```
Comment (JustinLin610) [inline models/ofa/unify_transformer_layer.py:125]: format

File: models/ofa/unify_transformer.py [hunk 3]
```diff
@@ -503,7 +608,7 @@ def __init__(self, args, dictionary, embed_tokens):
         self.entangle_position_embedding = args.entangle_position_embedding
 
     def build_encoder_layer(self, args, drop_path_rate=0.0):
-        layer = TransformerEncoderLayer(args, drop_path_rate=drop_path_rate)
+        layer = TransformerEncoderLayer(args, drop_path_rate=drop_path_rate,use_adapter=getattr(args, "adapter", False),adapter_dim=getattr(args, "adapter_dim", 200))
```
Comment (JustinLin610) [inline models/ofa/unify_transformer.py:611]: format

File: models/ofa/unify_transformer.py [hunk 4]
```diff
@@ -770,18 +903,28 @@ def forward_scriptable(
...
+                else:
+                    if idx == 0:
+                        prompt_kv = prompt_kv_list[idx]
+                    else:
+                        prompt_kv = None
+            else:
+                prompt_kv = None 
             x = layer(
-                x, encoder_padding_mask=encoder_padding_mask if has_pads else None, self_attn_bias=self_attn_bias
+                x, encoder_padding_mask=encoder_padding_mask if has_pads else None, self_attn_bias=self_attn_bias, prompt_kv=prompt_kv
```
Comment (JustinLin610) [inline models/ofa/unify_transformer.py:918]: format problem. make it multiple lines

File: README.md [hunk 5]
```diff
@@ -35,37 +23,38 @@ This source code is licensed under the Apache 2.0 license found in the LICENSE f
...
 * [Visual Question Answering](https://huggingface.co/spaces/OFA-Sys/OFA-Visual_Question_Answering)
+* [Text-to-Image Generation](https://huggingface.co/spaces/OFA-Sys/OFA-Text2Image_Generation)
+* [Generic Interface](https://huggingface.co/spaces/OFA-Sys/OFA-Generic_Interface)
 
 Also we provide Colab notebooks for you to better perceive the procedures. Click [here](colab.md) to check them out!
 <br></br>
 
 
 # News 
+* 2022.8.16: Released the **Chinese** version of OFA. **OFA-CN** needs only switching to `bpe_dir=../../utils/BERT_CN_dict` and using our provided Chinese checkpoints in [checkpoints_zh.md](checkpoints_cn.md). Temporarily, we only provide base-size and large-size pretrained checkpoints and finetuned checkpoints on [MUGE Caption](https://tianchi.aliyun.com/muge) and the Chinese version of RefCOCO(-/+/g) (to release soon).
```
Comment (JustinLin610) [inline README.md:45]: Update to the latest. I made mistake by using the old name checkpoints_zh.md, which should be checkpoints_cn.md

**Comment by yh351016** [comment]
All formatting issues have been corrected

File: models/ofa/unify_transformer_layer.py (hunk 2 above)
Comment (yh351016) [reply models/ofa/unify_transformer_layer.py]: done

**Pull request merged**